hack/update-schemas.sh

```diff
@@ -24,55 +24,42 @@ GOFLAGS=""
 CRD_PATH=$(dirname "${0}")/../config/300-crds
 API_PATH=$(dirname "${0}")/../pkg/apis
 
-TEMP_DIR_LOGS=$(mktemp -d)
-
-for FILENAME in `find $CRD_PATH -type f`;
-do
-  echo "Gererating CRD schema for $FILENAME"
-
-  # NOTE: APIs for the group tekton.dev are implemented under ./pkg/apis/pipeline,
-  # while the ResolutionRequest from group resolution.tekton.dev is implemented under ./pkg/apis/resolution
-
-  GROUP=$(grep -E '^  group:' $FILENAME)
-  GROUP=${GROUP#"  group: "}
-  if [ "$GROUP" = "tekton.dev" ]; then
-    API_SUBDIR='pipeline'
-  else
-    API_SUBDIR=${GROUP%".tekton.dev"}
-  fi
-
+mapfile -d '' FILES < <(find "$CRD_PATH" -type f -name '*.yaml' -print0)
+for file in "${FILES[@]}"; do
+  echo "Generating CRD schema for $file"
+  # NOTE: Workaround for https://github.com/kubernetes-sigs/controller-tools/pull/627
+  #
+  # Tekton splits its CRD definitions into multiple sub-packages. E.g. under
+  # the same `tekton.dev/v1alpha1` we find the following packages:
+  #  pkg/apis/
+  #  ├── pipeline
+  #  │   └── v1alpha1
+  #  ├── resource
+  #  │   └── v1alpha1
+  #  └── run
+  #      └── v1alpha1
+  # This breaks controller-gen's assumption of 1 group/version -> 1 go package.
+  # As a workaround, we patch every crd in isolation (tmp dir) with the correct
+  # API sub dir path.
   TEMP_DIR=$(mktemp -d)
-  cp -p $FILENAME $TEMP_DIR/.
-  LOG_FILE=$TEMP_DIR_LOGS/log-schema-generation-$(basename $FILENAME)
-    
-  counter=0 limit=10
-  while [ "$counter" -lt "$limit" ]; do
-    # FIXME:(burigolucas): add schema for fields with generic type once supported by controller-tools
-    # FIXME:(burigolucas): add schema for recursive fields once supported by controller-tools
-    # FIXME:(burigolucas): add reference for dependent external/internal schemas once supported in CRD specification
-    # FIXME:(burigolucas): controller-gen return status 1 with message "Error: not all generators ran successfully"
```

Comment:
I see that in the change, this FIXME isn't there, so has this one been addressed?

Reply:
@waveywaves Yes, the following has been resolved: `FIXME:(burigolucas): controller-gen return status 1 with message "Error: not all generators ran successfully"`.